github/prci.py

```diff
@@ -19,11 +22,76 @@
 )
 from internals.gql import util, queries
 
-
 logger = logging.getLogger(__name__)
 
 
 ERROR_BACKOFF_TIME = 600
+OS_PID_FILE_PATH = "/run/"
+OS_PID_FILE = os.path.join(OS_PID_FILE_PATH, "prci.pid")
+
+
+def pid_file_creation():
+    """
+        Creates PID file with Proccess ID as content.
+    """
+    with open(OS_PID_FILE, "w") as pid_file:
+        pid_file.write(str(os.getpid()))
+
+
+def pid_file_handling():
+    """
+        Create the PID file for each run.
+    """
+    if os.path.exists(OS_PID_FILE):
+        vagrant_process_prune()
+        vagrant_machines_prune()
+        pid_file_creation()
+
+    else:
+        pid_file_creation()
+
+
+def vagrant_process_prune():
+    """
+        Ensure that no 'vagrant up' process is running, 
+        to ensure a clean startup.
+    """
+    for process in psutil.process_iter():
+        if "ruby" in process.name():
+            for cmdline in process.cmdline():
+                if "vagrant" in cmdline:
+                    print(process) # change to logger info
+                    process.kill()
+
+
+def vagrant_machines_prune():
+    """
+        Prune all the vagrant machines in stuck state
+        from previous runs.
+    """
+
+    get_vagrant_machines = """
```

Comment:
Rename this to something that won't sound as a function/method, like: `get_vagrant_machines_command`.